providers/influxcloud/influxcloud.go

```diff
@@ -0,0 +1,150 @@
+// Package influxdata implements the OAuth2 protocol for authenticating users through InfluxCloud.
+// It is based off of the github implementation.
+package influxcloud
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strconv"
+
+	"github.com/markbates/goth"
+	"golang.org/x/oauth2"
+)
+
+const (
+	// The hard coded domain is difficult here because influx cloud has an acceptance
+	// domain that is different and we will need that for enterprise development.
+	authURL         string = "https://cloud.influxdata.com/oauth/authorize"
```

Comment:
I think to get started I'll submit the PR as is.

I'll open a different issue to discuss the alternative domains issue.